### ISSUE TITLE
NeuVector CRD deletion issue with foreground cascading deletion

### ISSUE THREAD

**Issue by minchao** [neuvector/neuvector#975]
**Environment**

Platform: Rancher
Kubernetes/Platform Version(s): v1.26.7-gke.500

**Describe the bug**

When using Argo CD to manage NeuVector's CRD, a problem arises during the deletion of CRD objects (e.g. NvSecurityRule). Despite being deleted by Argo CD, rules are shortly re-created after deletion.

Upon investigation, it was found that Argo CD uses [foreground cascading deletion as the default behavior](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#resources-prune-deletion-propagation-policy), whereas [kubectl defaults to background cascading deletion](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion).

In [foreground cascading deletion](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion), the Kubernetes API first adds the `.metadata.deletionTimestamp` and `.metadata.finalizers` fields to the object, placing the object into a "deletion in progress" state. The process awaits the GC to delete object dependencies. Afterward, the object is updated to [remove the `.metadata.finalizers` field](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/#how-finalizers-work). This update triggers a neuvector-validating-crd-webhook request, causing NeuVector to misconstrues this as a user trying to update an NvSecurityRule object.

A potential solution could involve NeuVector checking if the CRD object is in the "deletion in progress" state within the neuvector-validating-crd-webhook. If so, NeuVector should avoid re-creating the rule.

**To Reproduce**

1. Prepare NvSecurityRule CRD file `nv.helloworld-web.helloworld.yaml`.
2. Run `kubectl apply -f nv.helloworld-web.helloworld.yaml` to create a CRD object.
3. Confirm the added rule in the NeuVector UI.
4. Run `kubectl delete -f nv.helloworld-web.helloworld.yaml --cascade=foreground` to delete the CRD object using foreground cascading deletion. 
5. Confirm the deletion CRD object by running `kubectl get nvsecurityrule` (should already be deleted). 
6. NeuVector removes the rule.
7. NeuVector recreates the rule.
8. Check the NeuVector UI to see that the rule still exists.

**Expected behavior**

After deleting the NvSecurityRule CRD object, it should not exist within NeuVector.

**Screenshots**

As shown below, at 05:30, the NvSecurityRule CRD is deleted, but it is re-created 1 second later.

<img width="1182" alt="257191524-aa7319d2-ed0e-4c4a-a8d1-8e262d304bce" src="https://github.com/neuvector/neuvector/assets/3313639/026c29fc-1f28-4520-9f08-72ef76c196be">

**Comment by minchao** [comment]
A workaround method when managing CRD with Argo CD (GitOps):

```yaml
spec:
  syncPolicy:
    syncOptions:
    - PrunePropagationPolicy=background
```

**Comment by williamlin-suse** [comment]
This issue has been fixed in NV 5.3.3
